portfolio/src/main/java/com/google/sps/data/User.java

```diff
@@ -0,0 +1,136 @@
+        userEntity.setProperty(PHONE_PROPERTY, phone);
+        userEntity.setProperty(JOB_TITLE_PROPERTY, jobTitle);
+        return  userEntity;
+    }
+
+    public static boolean isUserDataSaved(String id){
+        Query commentsQuery = new Query(User.ENTITY_NAME).setFilter(new Query.FilterPredicate(User.ID_PROPERTY, Query.FilterOperator.EQUAL, id));
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        List<Entity> results = datastore.prepare(commentsQuery).asList(FetchOptions.Builder.withDefaults());
+        if(results.size()>0) return true;
+        else return false;
+    }
+
+    @Nullable
+    public static User getUserFromDatastore(String id){
+        return getUserFromUserEntity(getSavedUserEntity(id));
+    }
+
+    @Nullable
+    public static Entity getSavedUserEntity(String id){
+        Query commentsQuery = new Query(User.ENTITY_NAME).setFilter(new Query.FilterPredicate(User.ID_PROPERTY, Query.FilterOperator.EQUAL, id));
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        List<Entity> results = datastore.prepare(commentsQuery).asList(FetchOptions.Builder.withDefaults());
+        if(results.size()>0) return results.get(0);
+        else return null;
+    }
+
+    @Nullable
+    private static User getUserFromUserEntity(Entity userEntity){
+        if(userEntity==null) return null;
+        String id = (String) userEntity.getProperty(ID_PROPERTY);
+        String firstName = (String) userEntity.getProperty(FIRST_NAME_PROPERTY);
+        String lastName = (String) userEntity.getProperty(LAST_NAME_PROPERTY);
+        String email = (String) userEntity.getProperty(EMAIL_PROPERTY);
+        String phone = (String) userEntity.getProperty(PHONE_PROPERTY);
+        String jobTitle = (String) userEntity.getProperty(JOB_TITLE_PROPERTY);
+        return  new User(id, firstName, lastName, email, phone, jobTitle);
+    }
+
+    public static User getUserFromRequest(HttpServletRequest request){
+        String firstName = request.getParameter("first-name");
+        String lastName = request.getParameter("last-name");
+        String email = getUserEmailFromUserService();
+        String phone = request.getParameter("phone");
+        String jobTitle = null;
+        if(request.getParameterValues("type")!=null){ //box is checked
+            jobTitle = request.getParameter("job-title");
+        }
+        User newUser = new User(getUserIdFromUserService(), firstName, lastName, email, phone, jobTitle);
+        return newUser;
+    }
+
+    public static String getUserEmailFromUserService(){
+        UserService userService = UserServiceFactory.getUserService();
+        return userService.getCurrentUser().getEmail();
+    }
+
+    public static String getUserIdFromUserService(){
+        UserService userService = UserServiceFactory.getUserService();
+        return userService.getCurrentUser().getUserId();
+    }
```

Comment:
Although these methods are quite useful, they shouldn't be the responsibility of the "User" class. The class should only be responsible for holding data values. Otherwise this class is highly coupled with the datastore. If in the future you want to store the user somewhere else (or possibly multiple places during a staged code refactoring or testing), the User class shouldn't have to change. 
A better way would be to have something like `UserStorageManager` which could be an interface. Then you could have an implementation which could be `DataStoreUserStorageManager` and possibly other implementations in the future.

Reply:
This comment also applies to the `Comment` class.

Reply:
done for User class

Reply:
done for Comment class